subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/mvnsettings/DefaultLocalMavenRepositoryLocatorTest.groovy

```diff
@@ -97,25 +98,38 @@ class DefaultLocalMavenRepositoryLocatorTest extends Specification {
         locator.localMavenRepository == repo1
     }
 
-    def "ignores changes to maven settings file after initial load"() {
+    def "honors location specified in global settings file"() {
+        writeSettingsFile(locations.globalSettingsFile, repo1)
+
+        expect:
+        locator.localMavenRepository == repo1
+    }
+
+    def "returns default location if user settings file specifies empty local repository"() {
         when:
-        writeSettingsFile(locations.userSettingsFile, repo1)
+        writeSettingsFile(locations.userSettingsFile, "")
```

Comment:
Not sure if it makes a big difference but maybe we should also test `<localRepository/>` vs. `<localRepository><localRepository/>`.